dango/auth/src/lib.rs

```diff
@@ -91,13 +89,14 @@ pub fn authenticate_tx(
     };
 
     // Compute the sign bytes.
-    let sign_bytes = SignDoc {
-        messages: tx.msgs,
-        chain_id: ctx.chain_id,
-        sequence: metadata.sequence,
-    }
-    .to_json_vec()?
-    .hash256();
+    let sign_bytes = ctx.api.sha2_256(
+        &SignDoc {
+            messages: tx.msgs,
+            chain_id: ctx.chain_id,
+            sequence: metadata.sequence,
+        }
+        .to_json_vec()?,
+    );
```

Comment:
Use the host function for hashing instead. By removing the sha256 logic, contract sizes go down by ~10 KB.